### PR TITLE
QA-199 Matcher tests

### PR DESCRIPTION
**Pull request by madlabman**

### CODE REVIEW

File: tests/plugin/common.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+@contextmanager
+def fake_logger(mock_path, logger):
+    """Fake Allure logger"""
+
+    blocked_plugins = []
+    for name, plugin in allure_commons.plugin_manager.list_name_plugin():
+        allure_commons.plugin_manager.unregister(plugin=plugin, name=name)
+        blocked_plugins.append(plugin)
+
+    with mock.patch(mock_path) as ReporterMock:  # pylint: disable=invalid-name
```
Comment (dgusakov) [inline tests/plugin/common.py:33]: Naming convention violation. Use snake_case for variable naming

File: tests/plugin/test_matcher.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        TestCase("test_one_parameter_marked_only[1]"),
+        TestCase("test_one_parameter_marked_only[2]", matches=[simple_scenario]),
+    ), not_implemented
+
+
+@pytest.fixture()
+def collector_mock(scenarios: List[Scenario]) -> Type[Collector]:
+    """Return Collector mock class"""
+
+    class TestCollector(Collector):
```
Comment (dgusakov) [inline tests/plugin/test_matcher.py:100]:
```suggestion
    class DummySpecCollector(Collector):
```

**Review by dgusakov — CHANGES_REQUESTED** [review]
Remove binary files from the repo

File: tests/plugin/common.py (hunk 1 above)
Comment (madlabman) [reply tests/plugin/common.py:33]: Target of mock is a class object therefore I think it's better to keep CamelCase.

File: tests/plugin/test_matcher.py (hunk 2 above)
Comment (madlabman) [reply tests/plugin/test_matcher.py]: Renamed.

File: tests/plugin/test_matcher.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    with allure.step("Check fake test cases"):
+        for item in fake_items:
+            result = _has_result(allure_results.test_cases, item.name)
+            assert result["status"] == "unknown", "Unexpected entry status"
+            for match in item.matches:
+                _has_link(result, match.link)
+                _has_labels(result, custom_labels, match.specifications_names)
+                _has_labels(result, ScenariosMatcher.DEFAULT_LABELS, match.suites_names)
+
+    assert not allure_results.test_cases, "Unexpected test cases reported"
```
Comment (vgorkavenko) [inline tests/plugin/test_matcher.py:221]: I think it is not `AssertionError`. This arises when a developer make mistake (
Added some test cases, but did not process them here), right?

Comment (madlabman) [reply tests/plugin/test_matcher.py:221]: The main purpose of this statement is to check that we have only expected test runs in report.
This assert will fail in the case of developer mistake and it's okay, it is still assert and should raise AssertionError.

File: tests/plugin/test_matcher.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    test_cases: Tuple[Collection[TestCase], Collection[Scenario]],
+    scenarios: Collection[Scenario],
+    custom_labels: Collection[str],
+):
+    """Test matcher"""
+
+    cases, not_implemented = test_cases
+    pytester_result, allure_results = run_with_allure(
+        pytester=pytester,
+        testfile_path="test_matcher.py",
```
[the remark this answers is not in the document]
Comment (madlabman) [reply tests/plugin/test_matcher.py]: I've renamed file.

File: tests/plugin/test_matcher.py [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+def _conftest(pytester: Pytester, collector_mock: Type[Collector]) -> None:
+    pytest.CollectorMock = collector_mock
+    conftest = """'''Register Collector mock'''
+import pytest
+
+
+def pytest_register_spec_collectors(collectors) -> None:
+    collectors["test"] = pytest.CollectorMock
+    """
+    pytester.makeconftest(conftest)
```
[the remark this answers is not in the document]
Comment (vgorkavenko) [reply tests/plugin/test_matcher.py]: Now there are not "breaking" tabs and code is much easier to read

**Review by vgorkavenko — APPROVED** [review]

**Review by dgusakov — APPROVED** [review]

**Pull request merged**